### DISCUSSION TITLE
Deployment concerns

### DISCUSSION THREAD

**Discussion by bob-mcrae** [timeseriesAI/tsai#272] [Q&A] [answered]
We've used this excellent library for training using the InceptionTimePlus model and we're now looking to deploy saved models for inference in a production environment.  For this inference, all we need is to load the model & make predictions; basically two library commands.  The trouble I am currently facing is that it's taking a long time to load the library in what I thought was a minimal form.  For example, the following command is taking a full 45-seconds on my local PC and ~35 seconds in Google Colab.
`from tsai.all import load_learner`

I realize that this import will only need to occur once per service startup, but I was hoping to make the service as light-weight and fast as possible.  Are there any tips on improving the speed of what we're trying to do?  Thanks in advance.

**Comment by oguiza** [comment]
Hi @bob-mcrae,
Thanks for creating this discussion.
I've created an issue based on it ([#273](https://github.com/timeseriesAI/tsai/issues/273))

**Reply by bob-mcrae** [reply]
Thank you so much Ignacio!  Wow; very impressive and looks like a great implementation.  Unfortunately, I am unable to test -- probably a user problem.  I am getting a module not found error.

I installed 0.2.22 via: `!pip install tsai -U` within Colab.  I confirmed the version via `!pip show tsai`.  And checking the package directory, I am not seeing the file `inference.py`

**Reply by bob-mcrae** [reply]
Also, I just reproduced the issue on my local PC.

**Reply by oguiza** [reply]
Sorry, I forgot to mention that I've just created the new module in the new version which is still only available on GitHub. I don't know if you can install from there using: 

```bash
pip install git+https://github.com/timeseriesAI/tsai.git
```
I'm planning to create a new pip release within a day or two.

**Reply by bob-mcrae** [reply]
Trying now...

**Reply by bob-mcrae** [reply]
`from tsai.inference import load_learner` ::: 1.057 seconds!

**Reply by bob-mcrae** [reply]
...and, inference is working as expected.

VERY, VERY nice work Ignacio!  Thank you!

**Reply by bob-mcrae** [reply]
Hi Ignacio,

I am working on my docker image now.  Any idea on the timeline for the pip release that addressed this issue (273)?  Thanks in advance.

**Reply by oguiza** [reply]
Hi Bob, 
Either later today if I can find the time or tomorrow.

**Reply by oguiza** [reply]
Hi @bob-mcrae,
FYI, I've just created the new pip release (0.2.23).

**Reply by bob-mcrae** [reply]
Thanks so much Ignacio.  Great work!

**Comment by oguiza** [comment, marked as answer]
Great! I'm glad it works as expected! 😀

I'll close this issue now. Please, reopen it if necessary. 

And thanks a lot for bringing this up to my attention.
